### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by Tresor11**
on this branch we :

- created the post model
- created the user model
- added the validations
- implemented the sign_in and sign_up with devise
- filled the layout but not totally :smile:

### CODE REVIEW

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module UsersHelper
+  def my_posts
+    current_user.posts.all
+  end
+end
```
Comment (ebukaume) [inline app/helpers/users_helper.rb]: - [ ] There should be NO application logic in the views nor helpers.

You should move this to the user model as an instance method. Then in your controller, you initialize an instance variable to be used in your view.

**Comment by Tresor11** [comment]
thank you :smiley:  . we have added more tests to the rspec and removed the logic from the views

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  before :each do
+    @user = User.create(email: 'foobar@example.com', last_name: 'foobar', user_name: 'foobar', gender: 'male', first_name: 'foobar', password: 'foobar', password_confirmation: 'foobar')
```
Comment (mosaaleb) [inline spec/models/post_spec.rb]: - [ ] **[OPTOINAL]** I suggest that you use [`factory_bot`](https://github.com/thoughtbot/factory_bot) in the coming tests to clean your specs.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<div class="container-fluid posts">
+<div class="row">
+<div class="col-3 ">
+<%= render  "shared/new_post" %>
+</div>
+<div class="col-5 post-list">
+<div class="count text-center"> Recent posts <%=@posts.count%></div>
+<%@posts.each do |post|%>
```
Comment (mosaaleb) [inline app/views/posts/index.html.erb]: - [ ] **[OPTIONAL]** You can use [collection rendering](https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-collections) to dry your views. Also please ident your code to make it more readable.

**Pull request merged**